src/WordSearch/WordSearchSolver.ts

```diff
@@ -35,10 +36,23 @@ export class WordSearchSolver {
         throw new Error('Cannot find an empty string in the wordsearch');
       }
     }
-    return this.search();
   }
 
-  private search(): Result[] {
+  setGrid(matrix: string[][]) {
+    this._matrix = matrix;
+  }
+
+  setDirections(useDiagonal: boolean, useCardinal: boolean) {
+    this._directions = [];
+    if (useCardinal) {
```

Comment:
Should we throw if the user is asking for no directions?

Reply:
That's a good question - I thought about it, but ultimately decided against it.  It seems like a UI concern to me, since at least in theory, it's valid input.